nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/service/ColdStorageServiceImpl.java

```diff
@@ -54,84 +77,134 @@
 
-    private static final Logger log = LogManager.getLogger(ColdStorageHelper.class);
+    private static final Logger log = LogManager.getLogger(ColdStorageServiceImpl.class);
 
-    public static final String COLD_STORAGE_FACET_NAME = "ColdStorage";
-
-    public static final String FILE_CONTENT_PROPERTY = "file:content";
-
-    public static final String COLD_STORAGE_CONTENT_PROPERTY = "coldstorage:coldContent";
-
-    public static final String COLD_STORAGE_BEING_RETRIEVED_PROPERTY = "coldstorage:beingRetrieved";
+    /** @since 10.10 **/
+    protected static final List<String> COLD_STORAGE_DISABLED_RECOMPUTATION_LISTENERS = Arrays.asList(
+            UpdateThumbnailListener.THUMBNAIL_UPDATED, ThumbnailConstants.DISABLE_THUMBNAIL_COMPUTATION,
+            VideoChangedListener.DISABLE_VIDEO_CONVERSIONS_GENERATION_LISTENER,
+            PictureViewsGenerationListener.DISABLE_PICTURE_VIEWS_GENERATION_LISTENER);
 
     /** @since 10.10 **/
-    public static final String COLD_STORAGE_TO_BE_RESTORED_PROPERTY = "coldstorage:toBeRestored";
+    public static final String COLDSTORAGE_RENDITION_EP = "coldStorageRendition";
 
-    public static final String GET_DOCUMENTS_TO_CHECK_QUERY = String.format(
-            "SELECT * FROM Document, Relation WHERE %s = 1", COLD_STORAGE_BEING_RETRIEVED_PROPERTY);
+    /** @since 10.10 **/
+    protected static String defaultRendition;
 
     /** @since 10.10 **/
-    public static final String COLD_STORAGE_CONTENT_RESTORED_EVENT_NAME = "coldStorageContentRestored";
+    protected static final Map<String, String> renditionByDocType = new HashMap<>();
 
     /** @since 10.10 **/
-    public static final String COLD_STORAGE_CONTENT_TO_RESTORE_EVENT_NAME = "coldStorageContentToRestore";
+    protected static final Map<String, String> renditionByFacets = new HashMap<>();
 
-    public static final String COLD_STORAGE_CONTENT_AVAILABLE_EVENT_NAME = "coldStorageContentAvailable";
+    public ColdStorageServiceImpl() {
+        // no instance allowed
+    }
 
-    /** @since 10.10 **/
-    public static final String COLD_STORAGE_CONTENT_MOVED_EVENT_NAME = "coldStorageContentMoved";
+    @Override
+    public void registerContribution(Object contribution, String extensionPoint, ComponentInstance contributor) {
+        if (COLDSTORAGE_RENDITION_EP.equals(extensionPoint)) {
+            ColdStorageRenditionDescriptor descriptor = (ColdStorageRenditionDescriptor) contribution;
+            String docType = descriptor.getDocType();
+            if (docType != null) {
+                renditionByDocType.put(docType, descriptor.getRenditionName());
+            }
+            String facet = descriptor.getFacet();
+            if (facet != null) {
+                renditionByFacets.put(docType, descriptor.getRenditionName());
+            }
+            if (docType == null && facet == null) {
+                defaultRendition = descriptor.getRenditionName();
+            }
```

Comment:
It could be interesting to leverage the generic Descriptor registry here. You can have example around https://github.com/nuxeo/nuxeo/blob/10.10/nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkActionDescriptor.java.

Reply:
> It could be interesting to leverage the generic Descriptor registry here. You can have example around https://github.com/nuxeo/nuxeo/blob/10.10/nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkActionDescriptor.java.

I haven't seen this message (I'm on it)

Reply:
Leveraging `Descriptor` interface allows you to not override the `registerContribution`.

As you need to retrieve some information, you can do this in the `start` method with help of `DefaultComponent#getDescriptors(String)` or `DefaultComponent#getDescriptor(String)` method.

As it, contribution overriding will be handled for you.

Reply:
Hello,
 I got the first part, but for the second I don't.
When you say using the start method, you are referring to the constructor for example `ColdStorageServiceImpl`.

Reply:
thanks a lot

Reply:
yes It’s  more clear now
👍

Reply:
the stream looks very nice and cute. But if I have a choice I prefer it as it is ( I could maintain the code).

It may be an improvement when the team will be out of a rush.